.github/workflows/tox.yml

```diff
@@ -32,12 +31,14 @@ jobs:
     - name: Test with tox
       run: |
         pyenv="py$(echo "${{ matrix.python-version }}" | tr -d '.')"
-        tox -e ${pyenv},${{ matrix.tox-testenv }},flake8,lint
+        tox -e ${pyenv}-test,${pyenv}-rapidjson
+    - name: Check style
+      if: matrix.python-version == 3.9
+      run: tox -e flake8,lint
```

Comment:
Still would like to see the `Check style` check before `Test with tox`

Reply:
This is inconsistent with how Tox is configured currently, when run alone (e.g `tox`), it will run `test`, `rapidjson`, `lint` and `flake8`, in this order. The current change is consistent with that but run the lint/flake8 only for python 3.9.
See also https://github.com/RIOT-OS/riotctrl/pull/25#issuecomment-924046914

Reply:
You are reworking the whole CI anyways with this PR. So why stop at that?

Reply:
(and sorry somehow missed your comment upstairs)